src/fortrancode.l

```diff
@@ -1212,7 +1213,8 @@ static MemberDef *getFortranDefs(yyscan_t yyscanner,const QCString &memberName,
         { // found function in global scope
           if (cd == nullptr)
           { // Skip if bound to type
-            return md.get();
+            if (md.get()->getOutputFileBase() == yyextra->sourceFileDef->getOutputFileBase()) return md.get();
```

Comment:
I think the check should be `md->getFileDef()==yyextra->sourceFileDef` instead of comparing the output file, which may be different depending on the `SEPARATE_MEMBER_PAGES` setting, for instance.

Reply:
Might be the case, changed as indicated.